src/main/kotlin/no/nav/familie/ks/sak/kjerne/behandling/HenleggBehandlingService.kt

```diff
@@ -67,11 +69,25 @@ class HenleggBehandlingService(
         behandling.status = BehandlingStatus.AVSLUTTET
         behandling.aktiv = false
 
-        // oppdater fagsak status til Avsluttet
-        if (behandlingRepository.finnBehandlinger(behandling.fagsak.id).size == 1) {
-            fagsakService.oppdaterStatus(behandling.fagsak, FagsakStatus.AVSLUTTET)
+        // oppdater fagsak status til avsluttet hvis første behandling på fagsak er henlagt
+        val alleBehandlinger = behandlingRepository.finnBehandlinger(behandling.fagsak.id)
+        when {
+            alleBehandlinger.size == 1 -> {
+                fagsakService.oppdaterStatus(behandling.fagsak, FagsakStatus.AVSLUTTET)
+            }
+            else -> {
+                // aktiverer siste vedtatt behandling på nytt
+                alleBehandlinger.filter { !it.erHenlagt() && it.status == BehandlingStatus.AVSLUTTET }
+                    .maxByOrNull { it.opprettetTidspunkt }?.apply {
+                        aktiv = true
+                        behandlingRepository.saveAndFlush(this)
+                    }
+            }
         }
```

Comment:
```suggestion
        when (alleBehandlinger.size) {
            1 -> {
                fagsakService.oppdaterStatus(behandling.fagsak, FagsakStatus.AVSLUTTET)
            }
            else -> {
                // aktiverer siste vedtatt behandling på nytt
                alleBehandlinger.filter { !it.erHenlagt() && it.status == BehandlingStatus.AVSLUTTET }
                    .maxByOrNull { it.opprettetTidspunkt }?.apply {
                        aktiv = true
                        behandlingRepository.saveAndFlush(this)
                    }
            }
        }
```

Intellij klager på denne^